src/deploy.ts

```diff
@@ -77,30 +78,62 @@ type DeployTargetInfo =
   | {create: false; workspace: {id: string; login: string}; project: GetProjectResponse};
 
 /** Deploy a project to ObservableHQ */
-export async function deploy(
-  {config, message, force, deployPollInterval = DEPLOY_POLL_INTERVAL_MS, maxConcurrency}: DeployOptions,
-  effects = defaultEffects
-): Promise<void> {
-  const {clack} = effects;
-  Telemetry.record({event: "deploy", step: "start", force});
-  clack.intro(`${inverse(" observable deploy ")} ${faint(`v${process.env.npm_package_version}`)}`);
+export async function deploy(deployOptions: DeployOptions, effects = defaultEffects): Promise<void> {
+  Telemetry.record({event: "deploy", step: "start", force: deployOptions.force});
+  effects.clack.intro(`${inverse(" observable deploy ")} ${faint(`v${process.env.npm_package_version}`)}`);
 
-  let apiKey = await effects.getObservableApiKey(effects);
-  const apiClient = new ObservableApiClient(apiKey ? {apiKey, clack} : {clack});
-  const deployConfig = await effects.getDeployConfig(config.root);
+  const deployInfo = deployOptions.deployId 
+    ? await continueExistingDeploy(deployOptions, effects)
+    : await startNewDeploy(deployOptions, effects);
 
-  if (deployConfig.workspaceLogin && !deployConfig.workspaceLogin.match(/^@?[a-z0-9-]+$/)) {
-    throw new CliError(
-      `Found invalid workspace login in ${join(config.root, ".observablehq", "deploy.json")}: ${
-        deployConfig.workspaceLogin
-      }.`
-    );
-  }
-  if (deployConfig.projectSlug && !deployConfig.projectSlug.match(/^[a-z0-9-]+$/)) {
-    throw new CliError(
-      `Found invalid project slug in ${join(config.root, ".observablehq", "deploy.json")}: ${deployConfig.projectSlug}.`
-    );
-  }
+  effects.clack.outro(`Deployed project now visible at ${link(deployInfo.url)}`);
+  Telemetry.record({event: "deploy", step: "finish"});
+}
+
+async function continueExistingDeploy(
+  deployOptions: DeployOptions,
+  effects: DeployEffects
+): Promise<GetDeployResponse> {
+  const {apiClient} = await getApiClientAndCurrentUser(effects);
+
+  const deployId = deployOptions.deployId!;
```

Comment:
You could keep the type safety that extra parameter was giving you like this:/

```suggestion
async function continueExistingDeploy(
  deployOptions: DeployOptions & {deployId: string},
  effects: DeployEffects
): Promise<GetDeployResponse> {
  const {apiClient} = await getApiClientAndCurrentUser(effects);

  const {deployId} = deployOptions;
```

Reply:
ah nice, thanks - will switch to that